swarmrl/engine/espresso.py

```diff
@@ -428,6 +427,95 @@ def add_confining_walls(self, wall_type: int):
         # the wall itself has no radius, only the particle radius counts
         self.colloid_radius_register.update({wall_type: 0.0})
 
+    def add_maze(
+        self,
+        maze_wall_start_point: pint.Quantity,
+        maze_wall_end_point: pint.Quantity,
+        maze_type: int,
+        wall_thickness: pint.Quantity,
+    ):
+        """
+        User defined walls will interact with particles through WCA.
+        Is NOT communicated to the interaction models, though.
+        The walls have a large height resulting in 2D-walls in a 2D-simulation.
+        The actual height adapts to the chosen box size.
+        The shape of the underlying constraint is a square.
+
+        Parameters
+        ----------
+        maze_wall_start_point : pint.Quantity
+        list of lists (len()=2) with wall coordinates
+             [x_begin, y_begin]
+        maze_wall_end_point : pint.Quantity
+        list of lists (len()=2) with wall coordinates
+             [x_end, y_end]
+        maze_type : int
+            Wall interacts with particles, so it needs its own type.
+        wall_thickness: pint.Quantity
+            wall thickness
+
+        Returns
+        -------
+        """
+        maze_wall_start_point = maze_wall_start_point.m_as("sim_length")
+        maze_wall_end_point = maze_wall_end_point.m_as("sim_length")
+        wall_thickness = wall_thickness.m_as("sim_length")
+
+        if len(maze_wall_start_point) != len(maze_wall_end_point):
+            raise ValueError(
+                " Please double check your maze walls. There are more or less "
+                f" starting points {len(maze_wall_start_point)} than "
+                f" end points {len(maze_wall_end_point)}. They should be equal."
+            )
+
+        self._check_already_initialised()
+        if (
+            maze_type in self.colloid_radius_register.keys()
+            and self.colloid_radius_register[maze_type] != 0.0
```

Comment:
this line relies on an early abort of the and statement, bcs otherwise we access a key that does not exist in the second line.